earth_enterprise/src/fusion/portableglobe/servers/fileunpacker/shared/filewalker_unittest.cpp

```diff
@@ -0,0 +1,144 @@
+
+// Test classes for file walker package.
+class FileWalkerTest : public testing::Test {
+ protected:
+  std::string glb_file;
+  std::string glc_file;
+
+  virtual void SetUp() {
+    glb_file = "../../../fusion/portableglobe/servers/fileunpacker/test_data/test.glb";
+    glc_file = "../../../fusion/portableglobe/servers/fileunpacker/test_data/test.glc";
+  }
+
+  // Helper function to find values in lists
+  template<typename T>
+  bool Contains(std::list<T> & element_list, const T & element)
+  {
+	  auto it = std::find(element_list.begin(), element_list.end(), element);
+	  return it != element_list.end();
+  }
+
+  // Helper function to check if a file exists
+  bool FileExists(const char *filename)
+  {
+    std::ifstream ifile(filename);
+    return (bool)ifile;
+  }
+
+  /**
+    * Test that the contents of the test GLB have expected size and values
+    */
+  void TestGlbContents() {
+    EXPECT_TRUE(FileExists(glb_file.c_str()));
+
+    if (!FileExists(glb_file.c_str())) {
+      return;
+    }
+
+    PortableGlcReader reader(glb_file.c_str());
+    GlcUnpacker unpacker(reader, false, false);
+    std::list<std::string> files;
+    unpacker.MapFileWalker(0, [&] (int layer, const char* file_name) {
+      files.push_back(file_name);
+      return true;
+    });
+
+    EXPECT_FALSE(files.empty());
+
+    EXPECT_EQ(files.size(), 59);
+
+    EXPECT_FALSE(Contains(files, std::string("data")));
+
+    EXPECT_TRUE(Contains(files, std::string("data/index")));
+
+    EXPECT_TRUE(Contains(files, std::string("icons/773_l.png")));
+
+    files.clear();
+    unpacker.MapFileWalker(0, [&] (int layer, const char* file_name) {
+      files.push_back(file_name);
+      return false;
+    });
+
+    EXPECT_EQ(files.size(), 1);
+  }
+
+  /**
+    * Test that the contents of the test GLC layers have expected size and values
+    */
+  void TestGlcContents() {
+    EXPECT_TRUE(FileExists(glc_file.c_str()));
+
+    if (!FileExists(glc_file.c_str())) {
+      return;
+    }
+
+    PortableGlcReader reader(glc_file.c_str());
+    GlcUnpacker unpacker(reader, true, false);
+    std::map<int, std::list<std::string>> file_map;
+    unpacker.MapFileWalker([&] (int layer, const char* file_name) {
+      file_map[layer].push_back(file_name);
+      return true;
+    });
+
+    std::list<int> layer_ids;
+    for(auto const& pair: file_map) {
+      layer_ids.push_back(pair.first);
+    }
+
+    EXPECT_TRUE(Contains(layer_ids, 1));
+
+    EXPECT_TRUE(Contains(layer_ids, 2));
+
+    EXPECT_EQ(layer_ids.size(), 2);
+    
+    EXPECT_FALSE(Contains(file_map[1], std::string("data")));
+
+    EXPECT_FALSE(Contains(file_map[2], std::string("data")));
+
+    EXPECT_TRUE(Contains(file_map[1], std::string("icons/773_l.png")));
+
+    EXPECT_TRUE(Contains(file_map[2], std::string("icons/773_l.png")));
+
+    EXPECT_EQ(file_map[1].size(), 59);
+
+    EXPECT_EQ(file_map[2].size(), 52);
+  }
+};
+
+// Tests GLB and GLC directory contents
+TEST_F(FileWalkerTest, TestFiles) {
```

Comment:
Can we have three test and move these calls into there own test.  It just makes things more granular when things fail we have a more isolated place to look.  I was thinking three tests: `TestFilesGlobe`, `TestFilesComposite`, `TestFilesCompositeEarlyExit`.  Or something like that.